pyabc/petab/amici.py

```diff
@@ -0,0 +1,137 @@
+        Whether to estimate free parameters (column ESTIMATE=1 in the
+        parameters table).
+    fixed_parameters:
+        Whether to estimate fixed parameters (column ESTIMATE=0 in the
+        parameters table).
+    amici_model:
+        A corresponding compiled AMICI model that allows simulating data for
+        parameters. If not provided, one is created using
+        `amici.petab_import.import_petab_problem`.
+    amici_solver:
+        An AMICI model to simulate the model. If not provided, one is created
+        using `amici_model.getSolver()`.
+    store_simulations:
+        Whether to store performed simulations. Per default, only parameters
+        and likelihood valuaes are stored. Since an ODE model
+        is deterministic, the trajectories can easily be reproduced.
+    """
+
+    def __init__(
+            self,
+            petab_problem: petab.Problem,
+            amici_model: amici.Model = None,
+            amici_solver: amici.Solver = None,
+            free_parameters: bool = True,
+            fixed_parameters: bool = False,
+            store_simulations: bool = False):
+        super().__init__(
+            petab_problem=petab_problem,
+            free_parameters=free_parameters,
+            fixed_parameters=fixed_parameters)
+
+        if amici_model is None:
+            amici_model = amici.getab_import.import_petab_problem(
+                petab_problem)
+        self.amici_model = amici_model
+
+        if amici_solver is None:
+            amici_solver = self.amici_model.getSolver()
+        self.amici_solver = amici_solver
+
+        self.store_simulations = store_simulations
+
+    def create_model(
+        self
+    ) -> Callable[[Union[Sequence, Mapping]], Mapping]:
+        """Create model."""
+        # parameter ids to consider
+        x_ids = self.petab_problem.get_x_ids(
+            free=self.free_parameters, fixed=self.fixed_parameters)
+
+        # extract variables for improved pickling
+        petab_problem = self.petab_problem
+        amici_model = self.amici_model
+        amici_solver = self.amici_solver
+        store_simulations = self.store_simulations
+
+        # no gradients for pyabc
+        amici_solver.setSensitivityOrder(0)
+
+        def model(par: Union[Sequence, Mapping]) -> Mapping:
+            """The model function."""
+            # convenience to allow calling model not only with dicts
+            if not isinstance(par, Mapping):
+                par = {key: val for key, val in zip(x_ids, par)}
+
+            # simulate model
+            sim = simulate_petab(
+                petab_problem=petab_problem,
+                amici_model=amici_model,
+                solver=amici_solver,
+                problem_parameters=par,
+                scaled_parameters=True)
+
+            # return values of interest
+            ret = {'llh': sim['llh']}
+            if store_simulations:
+                for i_rdata, rdata in enumerate(ret['rdatas']):
```

Comment:
Replace rhs string literal by constants defined in AMICI?

Reply:
what would be the constant's name?

Reply:
I guess you should know :sweat_smile:  https://github.com/ICB-DCM/AMICI/pull/921/commits/3bac1bbabb785e9168dd0a8f8c993c6591650b43#diff-a1defc0ccc84ee9aa948637c02699a45R18

Reply:
uuuups yes of course :rofl: